core/common/src/main/java/alluxio/underfs/UnderFileStatus.java

```diff
@@ -66,6 +66,9 @@ public String toString() {
    * @return array of file or directory names
    */
   public static String[] toListingResult(UnderFileStatus[] children) {
+    if (children == null) {
+      return null;
```

Comment:
would it make sense to return an empty array instead? that way the caller does not have to test for null